src/annotator/anchoring/test/trim-range-test.js

```diff
@@ -0,0 +1,277 @@
+import { textNodes } from '../../../test-util/compare-dom';
+
+import { trimRange } from '../trim-range';
+
+import { TextRange } from '../text-range';
+
+describe('annotator/anchoring/trim-range', () => {
+  let container;
+
+  const htmlContent = `
+  <div id="A">
+    <div id="B">&#8233;z<p id="C"><span id="D">&nbsp; charm&nbsp;</span> a </p> </div>
+    <div id="E">
+      <p id="F"> <span id="G">&nbsp;<span id="H"><!-- comment -->f </span></span></p>
+    </div>
+    <div id="I"></div>
+  </div>
+  <div id="J"> <span id="K"> t </span><div id="L"></div><div id="M">&nbsp;<div> </div>
+        `;
+
+  afterEach(() => {
+    container?.remove();
+  });
+
+  const addContent = innerHTML => {
+    container = document.createElement('div');
+    container.innerHTML = innerHTML;
+    document.body.appendChild(container);
+    return container;
+  };
+
+  const addTextContent = text => {
+    const wrappingNode = document.createElement('p');
+    wrappingNode.innerHTML = text;
+    document.body.appendChild(wrappingNode);
+    return wrappingNode;
+  };
+
+  /**
+   * Return a Range that selects text content in `container` from `startOffset`
+   * to `endOffset`. Will select entire textContent of `container` if no offsets
+   * are provided.
+   *
+   * @param {Element} container
+   * @param {number} [startOffset] - or 0 if not present
+   * @param {number} [endOffset ] - or length of text content if not present
+   */
+  const rangeFromOffsets = (container, startOffset, endOffset) => {
+    return TextRange.fromOffsets(
+      container,
+      startOffset ?? 0,
+      endOffset ?? container.textContent.length
+    ).toRange();
+  };
+
+  /**
+   * Return a Range that starts before `startElement` and ends after
+   * `endElement`. Use `TextRange` to convert it such that start and end
+   * containers are guaranteed to be text nodes (but not necessarily the correct
+   * ones, per trimming).
```

Comment:
I would drop the parenthesized comment here, since what is expected/correct depends on the context in which the function is used.